src/coreclr/vm/readytoruninfo.h

```diff
@@ -48,6 +48,42 @@ class ReadyToRunCoreInfo
 typedef DPTR(class ReadyToRunInfo) PTR_ReadyToRunInfo;
 typedef DPTR(class ReadyToRunCoreInfo) PTR_ReadyToRunCoreInfo;
 
+
+class ReadyToRun_EnclosingTypeMap
```

Comment:
If I'm reading how this works correctly, these are partial types, right? As in they represent a collection and are cast to a pointer and then indexed into. I get this from looking at the implementation of `GetTypeGenericInfo()`.

My suggestion here would be not make the type that has the logic be the data too. Instead create a struct that is the data and then use it. For example, I think for `ReadyToRun_TypeGenericInfoMap` the data generally looks like the following:

```cpp
struct ReadyToRun_TypeGenericInfoMap_Data
{
    uint32_t TypeCount;
    uint8_t Data[ANYSIZE_ARRAY];
};
```

The logic for using the index from a RID makes sense, but then you check the lower bit after dividing it by 2. Is this an odd/even thing? I will comment there.

Reply:
Seems like a good plan, to be honest, this code ended up kinda gross.